src/Umbraco.Web/Editors/MediaController.cs

```diff
@@ -175,7 +176,37 @@ public IEnumerable<MediaItemDisplay> GetByIds([FromUri]int[] ids)
         /// Returns the child media objects
         /// </summary>
         [FilterAllowedOutgoingMedia(typeof(IEnumerable<ContentItemBasic<ContentPropertyBasic, IMedia>>), "Items")]
-        public PagedResult<ContentItemBasic<ContentPropertyBasic, IMedia>> GetChildren(int id,
+        public PagedResult<ContentItemBasic<ContentPropertyBasic, IMedia>> GetChildren(string id,
+            int pageNumber = 0,
+            int pageSize = 0,
+            string orderBy = "SortOrder",
+            Direction orderDirection = Direction.Ascending,
+            bool orderBySystemField = true,
+            string filter = "")
+        {
+            int idInt; Guid idGuid;
+
+            if (Guid.TryParse(id, out idGuid))
+            {
+                var entity = Services.EntityService.GetByKey(idGuid);
+                if (entity != null)
+                {
+                    return getChildren(entity.Id, pageNumber, pageSize, orderBy, orderDirection, orderBySystemField, filter);
+                }
+                else
+                {
+                    throw new HttpResponseException(HttpStatusCode.NotFound);
+                }
+            }
+            else if (int.TryParse(id, out idInt))
+            {
+                return getChildren(idInt, pageNumber, pageSize, orderBy, orderDirection, orderBySystemField, filter);
+            }
+
+            throw new HttpResponseException(HttpStatusCode.NotFound);
+        }
+
+        private PagedResult<ContentItemBasic<ContentPropertyBasic, IMedia>> getChildren(int id,

```

Comment:
Not sure why the GitHub diff here has not been updated @Shazwazza but updated the method name to `GetChildren` whilst I was doing the change for the localStorage clearing on login